### ISSUE TITLE
npt even with -K can stop running with an uncaught exception

### ISSUE THREAD

**Issue by cconstab** [atsign-foundation/noports#1231]
### Describe the bug

npt has an un caught exception and exit when it should be able to handle network errors

### Steps to reproduce

1. Run npt to a machine
2. kill the ethernet/wifi on that machine
3. see exception

### Expected behavior

npt always stays alive dispite tough network connections

### Screenshots

```
╰$ npt -f @cconstab -t @ssh_1 -r @rv_am -d cally -l 1883 -p 1883  -T 180m -K
Connecting ... Connected
2024-07-30 19:10:29.345249 : Sending daemon feature check request
2024-07-30 19:10:29.345388 : Fetching host and port from srvd
2024-07-30 19:10:33.498013 : Received host and port from srvd
2024-07-30 19:10:33.498071 : Waiting for daemon feature check response
2024-07-30 19:10:33.498085 : Received daemon feature check response
2024-07-30 19:10:33.500420 : Required daemon features are supported
2024-07-30 19:10:33.684594 : Sending session request to the device daemon
2024-07-30 19:10:33.892085 : Waiting for response from the device daemon
2024-07-30 19:10:34.810276 : Received response from the device daemon
2024-07-30 19:10:34.816711 : Will use local port 1883
2024-07-30 19:10:34.816735 : Creating connection to socket rendezvous
2024-07-30 19:10:34.848773 : npt is listening on localhost:1883

Error: SocketException: Connection reset by peer (OS Error: Connection reset by peer, errno = 54), address = 449376cb-c771-5449-9645-0cf05a82195e.swarm0002.atsign.zone, port = 61758

Stack Trace:
2024-07-30 19:17:56.149994 : Cleaning up temporary files
╭─cconstab@tarial in ~
╰$ npt
Version : 5.5.0 (core: 6.1.0)
```

### Smartphones

- npt/sshnpd 5.5.0 

### Were you using an atApplication when the bug was found?

npt

**Comment by gkc** [comment]
@cconstab Which branch did you test with? This new ticket you've created is I think an exact duplicate of #1182 for which there is a proposed fix in #1207

**Comment by cconstab** [comment]
Doh .. I had thought this was merged
